### DISCUSSION TITLE
coalesce in mutate

### DISCUSSION THREAD

**Discussion by alexandro-ag** [markfairbanks/tidypolars#161] [Q&A]
I looked through the documentation and did not find a ```coalesce``` function. Is there some way I can mimic this behavior given the current toolkit, specifically in a ```mutate``` context?

Example:

``` python
null_df = tp.from_pandas(pd.DataFrame({'a': [1], 'b': [2], 'c': [np.NaN]})) # didn't know how else to produce a null value
┌─────┬─────┬──────┐
│ a   ┆ b   ┆ c    │
│ --- ┆ --- ┆ ---  │
│ i64 ┆ i64 ┆ f64  │
╞═════╪═════╪══════╡
│ 1   ┆ 2   ┆ null │
└─────┴─────┴──────┘

null_df\
    .mutate(coalesced_cols = tp.coalesce(col('c'), col('b'), col('a'))) # expected value: 2 since order is c, b, a
┌─────┬─────┬──────┬────────────────┐
│ a   ┆ b   ┆ c    ┆ coalesced_cols │
│ --- ┆ --- ┆ ---  ┆ ---            │
│ i64 ┆ i64 ┆ f64  ┆ i32            │
╞═════╪═════╪══════╪════════════════╡
│ 1   ┆ 2   ┆ null ┆ 2              │
└─────┴─────┴──────┴────────────────┘
```

Edit:

Just thought it through. I can probably use an ```if_else``` for now?

Edit 2:

tested and it worked using ```if_else```. I'm now wondering what makes ```coalesce``` special over doing an ```if_else``` or ```case_when```

**Comment by markfairbanks** [comment]
Like you mentioned - for now I would recommend using `if_else()` or `case_when()`.

I'll have to think about whether this one is worth implementing. If we do implement it it will probably just call `if_else()`/`case_when()` in the background.

In the meantime I opened #162 as a feature request.
